### PR TITLE
Add get user agent functionality.

### PR DESCRIPTION
**Pull request by txdv**
When exposing the **set user agent** functionality in the [C# binding](https://github.com/libgit2/libgit2sharp/pull/1334) we saw that we can't really write a good test for whether we set it correctly since getting the set user agent is not possible.

This adds the possibility to get the user agent.

### CODE REVIEW

File: tests/core/useragent.c [hunk 1]
```diff
@@ -4,8 +4,12 @@
...
 {
 	const char *custom_name = "super duper git";
+	git_buf buf = GIT_BUF_INIT;
 
 	cl_assert_equal_p(NULL, git_libgit2__user_agent());
 	cl_git_pass(git_libgit2_opts(GIT_OPT_SET_USER_AGENT, custom_name));
 	cl_assert_equal_s(custom_name, git_libgit2__user_agent());
+
+	git_libgit2_opts(GIT_OPT_GET_USER_AGENT, &buf);
+	cl_assert_equal_s(custom_name, buf.ptr);
```
Comment (ethomson) [inline tests/core/useragent.c]: Please add a `git_buf_free(&buf)` here to free the memory.

File: tests/core/useragent.c [hunk 2]
```diff
@@ -4,8 +4,12 @@
 void test_core_useragent__get(void)
 {
 	const char *custom_name = "super duper git";
+	git_buf buf = GIT_BUF_INIT;
 
 	cl_assert_equal_p(NULL, git_libgit2__user_agent());
 	cl_git_pass(git_libgit2_opts(GIT_OPT_SET_USER_AGENT, custom_name));
 	cl_assert_equal_s(custom_name, git_libgit2__user_agent());
+
+	git_libgit2_opts(GIT_OPT_GET_USER_AGENT, &buf);
```
Comment (ethomson) [inline tests/core/useragent.c]: Please check that this succeeds - eg `cl_git_pass(git_libgit2_opts(...))`...

**Comment by txdv** [comment]
I added the freeing at the end, not directly on the line, so it frees but the comment is still shown.

**Comment by ethomson** [comment]
Perfect, can you squash this down into a single commit and then I'll merge it!  😀

**Comment by txdv** [comment]
Your wish is my desire. Is there a place to document this added functionality?

**Comment by ethomson** [comment]
Yes, thanks for reminding me!  Please add this to the `CHANGELOG.md`.  :D

File: CHANGELOG.md [hunk 3]
```diff
@@ -12,6 +12,10 @@ v0.24 + 1
 
 ### API additions
 
+* You can now get the user-agent used by libgit2 using the
+  `GIT_OPT_GET_USER_AGENT` option with `git_libgit2_opts()`.
+  It is the counterpart to `GIT_OPT_SET_USER_AGENT`.
```
Comment (ethomson) [inline CHANGELOG.md]: 👍

**Comment by ethomson** [comment]
Awesome, thanks again!  ✨

**Pull request merged**